library/src/scala/quoted/Quotes.scala

```diff
@@ -4254,8 +4254,27 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
       *   -  ...
       *   -  Nth element is `FunctionN`
       */
+      // TODO: deprecate in 3.4 and stabilize FunctionClass(Int)/ContextFunctionClass(Int)
+      // @deprecated("Use overload of `FunctionClass` with 1 or 2 arguments","3.4")
       def FunctionClass(arity: Int, isImplicit: Boolean = false, isErased: Boolean = false): Symbol
 
+      /** Class symbol of a function class `scala.FunctionN`.
+       *
+       *  @param arity the arity of the function where `0 <= arity`
+       *  @return class symbol of `scala.FunctionN` where `N == arity`
+       */
+      @experimental
+      def FunctionClass(arity: Int): Symbol
+
+      /** Class symbol of a context function class `scala.ContextFunctionN`.
```

Comment:
Yes, it is to keep source compatibility on `defn.FunctionClass(arity, isContextual)`.

I did use `isContextual` instead of `isImplicit` to make the API clearer eventhough it does not help with source compatibility. This is only source incompatible if they write the argument name explicitly.

Doc updated